codec-redis/src/main/java/io/netty/handler/codec/redis/RedisDecoder.java

```diff
@@ -126,7 +146,18 @@ private boolean decodeType(ByteBuf in) throws Exception {
         if (!in.isReadable()) {
             return false;
         }
+
+        final int initialIndex = in.readerIndex();
         type = RedisMessageType.valueOf(in.readByte());
+
+        if (type == RedisMessageType.INLINE_COMMAND && !decodeInlineCommands) {
+            throw new RedisCodecException("Decoding of inline commands is disabled");
+        }
+
+        if (type.length() == 0) {
```

Comment:
the `length()` method isn't really used in a generic fashion here and is a bit misleading. For example we always read 1 `byte` above without considering `length()`. Can we instead handle this `INLINE_COMMAND` logic as a continuation of the special case above?

Reply:
I'll move the entire type detection to `RedisMessageType`. This will also improve functional consistency with the suggested serialization change below.